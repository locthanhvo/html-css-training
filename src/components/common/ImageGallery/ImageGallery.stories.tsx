import { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';

// Components
import ImageGallery from '.';

export default {
  title: 'Components/ImageGallery',
  component: ImageGallery,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    ),
  ],
  parameters: {
    layout: 'centered',
  },
} as Meta;

type Story = StoryObj<typeof ImageGallery>;

export const Default: Story = {
  args: {
    previewURL: 'https://images.unsplash.com/photo-1555041469-a586c61ea9bc',
  },
};
