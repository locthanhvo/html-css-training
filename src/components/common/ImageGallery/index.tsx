import { memo } from 'react';
import { Box, Image } from '@chakra-ui/react';

interface Props {
  previewURL: string;
  index: number;
}

const ImageGallery = ({ previewURL }: Props) => {
  return (
    <Box position="relative">
      <Image
        w={230}
        src={previewURL}
        alt={previewURL}
        fallbackSrc={previewURL}
        objectFit="contain"
      />
    </Box>
  );
};

export default memo(ImageGallery);
