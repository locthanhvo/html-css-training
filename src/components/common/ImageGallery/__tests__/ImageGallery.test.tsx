import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

// Components
import ImageGallery from '..';

describe('ImageGallery Component', () => {
  it('should match snapshot', () => {
    const element = render(
      <BrowserRouter>
        <ImageGallery
          index={1}
          previewURL="https://images.unsplash.com/photo-1555041469-a586c61ea9bc"
        />
      </BrowserRouter>,
    );

    expect(element).toMatchSnapshot();
  });
});
