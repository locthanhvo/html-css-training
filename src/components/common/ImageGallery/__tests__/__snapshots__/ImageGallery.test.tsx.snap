// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageGallery Component should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-79elbk"
      >
        <img
          alt="https://images.unsplash.com/photo-1555041469-a586c61ea9bc"
          class="chakra-image__placeholder css-106i85g"
          src="https://images.unsplash.com/photo-1555041469-a586c61ea9bc"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-79elbk"
    >
      <img
        alt="https://images.unsplash.com/photo-1555041469-a586c61ea9bc"
        class="chakra-image__placeholder css-106i85g"
        src="https://images.unsplash.com/photo-1555041469-a586c61ea9bc"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
